File: app/src/main/java/com/example/studymanager/models/domain/StudieTask.kt
package com.example.studymanager.domain

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey

@Entity(
    tableName = "studie_task_table", foreignKeys = arrayOf(
        ForeignKey(
            entity = StudieVak::class,
            parentColumns = arrayOf("studieVakId"),
            childColumns = arrayOf("vakId"),
            onDelete = CASCADE
        )
    )
)
data class StudieTask(
    @PrimaryKey(autoGenerate = true)
    var studieTaskId: Int,
    val studieTaskTitle: String,
    val totalTaskDuration: Long,
    var remainingTaskTime: Long,
    var vakId: Int,
    var vakName: String
)

File: Backend/StudyManagerApi/Models/Domain/StudieTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StudyAPI.Models.Domain {
    public class StudieTask {

        public int StudieTaskId { get; set; }
        public string StudieTaskTitle { get; set; }
        public long TotalTaskDuration { get; set; }
        public long RemainingTaskTime { get; set; }
        public int VakId { get; set; }
        public string VakName { get; set; }
        public StudieVak vak { get; set; }
        public int GebruikerId { get; set; }

        public StudieTask() {

        }
        public StudieTask(string taskTitle, long totalDur,long remDur, string vakName,int gebruikerId) {
            this.StudieTaskTitle = taskTitle;
            this.TotalTaskDuration = totalDur;
            this.RemainingTaskTime = remDur;
            this.VakName = vakName;
            this.GebruikerId = gebruikerId;
        }
    }
}
